pkg/operation/botanist/health_check.go

```diff
@@ -506,7 +494,7 @@ func (b *Botanist) checkControlPlane(
 		return exitCondition, err
 	}
 	if gardenletfeatures.FeatureGate.Enabled(features.Logging) {
-		if exitCondition, err := checker.CheckLoggingControlPlane(b.Shoot.SeedNamespace, b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting, condition, seedDeploymentLister, seedStatefulSetLister); err != nil || exitCondition != nil {
+		if exitCondition, err := checker.CheckLoggingControlPlane(b.Shoot.SeedNamespace, b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting, condition, seedStatefulSetLister); err != nil || exitCondition != nil {
```

Comment:
There are landscapes where shoots are reconciled only in their maintenance time windows. e.g. once every 24 hours. Such shoots will be reported with failing controlplane till they are reconciled 
```
    Last Transition Time:  2020-07-21T18:50:02Z
    Last Update Time:      2020-07-21T19:11:41Z
    Message:               Missing required stateful sets: [loki]
    Reason:                StatefulSetMissing
    Status:                False
    Type:                  ControlPlaneHealthy
```

Could you enhance the check or temporary do not check for `loki`?

Reply:
Even after the shoot was reconciled and `loki` was deployed I still observing this failing condition.

Reply:
You could do the health check only if the shoot was reconciled by version of gardener that is using Loki, e.g. `shoot.status.gardener.version is >=v1.8.x`